### DISCUSSION TITLE
How to run a custom command only during a retry attempt?

### DISCUSSION THREAD

**Discussion by felipe969** [cypress-io/cypress#15456] [General]
Here is the thing, in the project that I am working on, I would like to run a resetDB command only when necessary instead of executing every spec on hooks (e.g when a test fails and a retry is performed). I haven't found anything on cypress runners that could be used on this task. 

Currently, I am using a great solution from this thread to not upload test videos on CI when test success. 

- https://github.com/cypress-io/cypress/issues/2522#issuecomment-749316813
- https://github.com/cypress-io/cypress/issues/6665

What I am looking for is something like these experimental events. I haven't found retries or attempts as an event listener option and the after in this case cant consider the attempts because it will not be an end process like the result before finishing all attempts. 

```
 on('after:spec', (spec, results) => {
    if (results.stats.retries) {
       // custom command 
    }
  }) 
```

Am I missing something that we already have? 

Thanks in advance.

**Comment by herecydev** [comment]
Have you seen https://docs.cypress.io/api/commands/task.html#Syntax

I feel like thats what you're asking to do if I understood you correctly.

**Comment by felipe969** [comment]
Hey, @herecydev tks for the reply. Actually, I am using the cy.task() for some routines. The problem regarding the opened thread is in how to identify when a retry attempt of failing testing is being executed and then do some action right after that attempt, in my example, a cleanDB method (already implemented).

**Reply by herecydev** [reply]
In which case there is:

```js
  // in the test
  const currentRetry = cy.state("runnable")._currentRetry
  cy.task("cleanDB", { currentRetry });
```

**Reply by felipe969** [reply]
@herecydev Hey many thanks, it helped. I just changed a little bit using cucumber before hooks checking its value on every run (if is greater than 0), it is working pretty well. The only bad side is that this value is being compared on every run, the ideal might be something added in the event listeners, catching the state of a retry attempt outside current test runner context. Thanks again
